core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -642,17 +642,17 @@ class LogCleanerTest extends JUnitSuite {
     }
 
     // grouping by very large values should result in a single group with all the segments in it
-    var groups = cleaner.groupSegmentsBySize(log.logSegments, maxSize = Int.MaxValue, maxIndexSize = Int.MaxValue)
+    var groups = cleaner.groupSegmentsBySize(log.logSegments, maxSize = Int.MaxValue, maxIndexSize = Int.MaxValue, log.logEndOffset + 1)
```

Comment:
log.logEndOffset is the next offset in the log. So, we can just pass in log.logEndOffset, instead of log.logEndOffset + 1. Ditto below.